### PR TITLE
Improve exception messages

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java [hunk 1]
```diff
@@ -275,12 +256,7 @@ public void run() {
                 try {
                   chunkData = readRequestData();
                 } catch (IOException e) {
-                  error =
-                      new RuntimeException(
-                          "InsertChunkResponseObserver.beforeStart for "
-                              + googleCloudStorageGrpcWriteChannel
-                              + " caught ",
-                          e);
+                  error = new IOException(String.format("Failed to for '%s'", resourceId), e);
```
Comment (hongyegong) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java]: Failed to request for?

File: util/src/main/java/com/google/cloud/hadoop/util/AsyncWriteChannelOptions.java [hunk 2]
```diff
@@ -62,6 +62,10 @@ public static Builder builder() {
 
   public abstract boolean isDirectUploadEnabled();
 
+  /**
```
Comment (hongyegong) [inline util/src/main/java/com/google/cloud/hadoop/util/AsyncWriteChannelOptions.java]: Doesn't look like a right place to put this document.

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java [hunk 3]
```diff
@@ -65,36 +67,32 @@
...
-      Map<String, String> objectMetadata,
+      Map<String, String> metadata,
       String contentType) {
     super(threadPool, options);
     this.stub = stub;
+    this.resourceId = resourceId;
     this.writeConditions = writeConditions;
     this.requesterPaysProject = requesterPaysProject;
+    this.metadata = ImmutableMap.copyOf(metadata);
+    this.contentType = contentType;
```
Comment (hongyegong) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java]: contentType is not declared above.

Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java]: It's declared in super class.

File: util/src/main/java/com/google/cloud/hadoop/util/AsyncWriteChannelOptions.java (hunk 2 above)
Comment (medb) [reply util/src/main/java/com/google/cloud/hadoop/util/AsyncWriteChannelOptions.java]: Done.

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java (hunk 1 above)
Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java]: Done.

**Pull request merged**